src/core/security/auth.c

```diff
@@ -44,12 +44,15 @@
 #include "src/core/security/credentials.h"
 #include "src/core/surface/call.h"
 
+#define MAX_CREDENTIAL_METADATA_COUNT 4
```

Comment:
I actually suspect we should move it back to 2 then... and up it when we actually have cause to up it.

Adding a new cred type is going to be a recompile anyway...